### PR TITLE
fix: apply forward threshold to send_message_to_user

### LINKED ISSUE #8678: [Bug]send_message_to_user 工具调用不触发合并转发消息
### What happened / 发生了什么

当 LLM 通过普通文本回复时，若回复字数超过 `platform_settings.forward_threshold`（默认 1500 字），AstrBot 会自动将消息包装为 QQ 合并转发消息（`send_group_forward_msg`）以防止刷屏。

但当 LLM 调用内置工具 `send_message_to_user` 发送消息时，无论内容多长，都不会触发合并转发，消息会以普通消息形式直接发出。

### Reproduce / 如何复现？

1. 使用 aiocqhttp（OneBot V11）平台适配器
2. 将 `platform_settings.forward_threshold` 设置为较小的值（如 100）
3. 让 LLM 调用 `send_message_to_user` 工具发送超过阈值字数的纯文本内容
4. 观察消息以普通消息发出，而非合并转发

### AstrBot version, deployment method (e.g., Windows Docker Desktop deployment), provider used, and messaging platform used. / AstrBot 版本、部署方式（如 Windows Docker Desktop 部署）、使用的提供商、使用的消息平台适配器

winnnn

### OS

Windows

### Logs / 报错日志

none

### Are you willing to submit a PR? / 你愿意提交 PR 吗？

- [ ] Yes!

### Code of Conduct

- [x] I have read and agree to abide by the project's [Code of Conduct](https://docs.github.com/zh/site-policy/github-terms/github-community-code-of-conduct)。

### PR DESCRIPTION
**Pull request by Last-emo-boy**
Fixes #8678.

### Modifications / 改动点

- Apply the existing `platform_settings.forward_threshold` rule to `send_message_to_user` output.
- Limit conversion to messages sent from an `aiocqhttp` event through the same adapter instance, where a valid bot UIN is available.
- Preserve plain-message behavior for cron, cross-platform, cross-instance, and non-`aiocqhttp` sends.
- Keep current-session duplicate reply suppression unchanged by recording plain text before wrapping the message in a forward node.

- [x] This is NOT a breaking change. / 这不是一个破坏性变更。

### Screenshots or Test Results / 运行截图或测试结果

Verification:

- `uv run pytest tests/unit/test_message_tools.py tests/unit/test_func_tool_manager.py tests/unit/test_astr_agent_tool_exec.py -q` — 50 passed.
- `uv run ruff format --check .` — 480 files already formatted.
- `uv run ruff check .` — passed.
- `git diff --check origin/master` — passed.

Regression coverage includes the strict `>` boundary, same-instance forwarding, current-session deduplication, and the cron/cross-platform fallback to plain messages.

---

### Checklist / 检查清单

- [x] This is a bug fix and does not add a new feature.
- [x] The change is covered by targeted regression tests.
- [x] No new dependency is introduced.
- [x] This change does not introduce malicious code.

## Summary by Sourcery

Apply QQ forward-threshold handling to SendMessageToUserTool while preserving existing plain-message behavior and current-session deduplication.

Bug Fixes:
- Ensure long aiocqhttp tool messages are wrapped into merged-forward nodes according to platform forward_threshold settings.
- Prevent creation of invalid merged-forward nodes for cross-platform or cross-instance sends, falling back to plain messages instead.
- Maintain correct duplicate-reply suppression by recording the original plain text before wrapping messages in forward nodes.

Tests:
- Add unit tests covering aiocqhttp forward-threshold application, boundary behavior at the threshold, and cross-platform behavior for SendMessageToUserTool.

### CODE REVIEW

File: astrbot/core/tools/message_tools.py [hunk 1]
```diff
@@ -332,10 +332,36 @@ async def call(
...
+            target_session.platform_id
+        )
+        if (
+            target_platform
+            and target_platform.meta().name == "aiocqhttp"
+            and context.context.event.get_platform_name() == "aiocqhttp"
+            and context.context.event.get_platform_id() == target_session.platform_id
+        ):
+            cfg = context.context.context.get_config(umo=str(target_session))
+            threshold = cfg.get("platform_settings", {}).get("forward_threshold", 1500)
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply astrbot/core/tools/message_tools.py:348]: Context.get_config has a non-optional AstrBotConfig contract and the config manager guarantees defaults. The forward_threshold schema is an integer, and ResultDecorateStage reads the same value directly. Adding local None handling or coercion here would make the two paths inconsistent and mask invalid configuration, so this keeps the established contract.

File: tests/unit/test_message_tools.py [hunk 2]
```diff
@@ -153,6 +162,87 @@ async def test_send_message_defaults_to_current_session():
...
+async def test_send_message_keeps_aiocqhttp_text_at_forward_threshold():
+    """Text equal to the threshold keeps the existing plain-message behavior."""
+    tool = SendMessageToUserTool()
+    ctx = _make_context(current_session="napcat:GroupMessage:123456")
+    ctx.context.context.get_platform_inst = lambda _platform_id: SimpleNamespace(
+        meta=lambda: SimpleNamespace(name="aiocqhttp")
+    )
+    ctx.context.context.get_config = lambda umo: {
+        "platform_settings": {"forward_threshold": 5}
+    }
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply tests/unit/test_message_tools.py:209]: Fixed in ac70a21f4 by mocking the source platform as aiocqhttp, so the strict threshold boundary now executes the production branch. The complete targeted set passes with 50 tests.

**Comment by Soulter** [comment]
thanks for your contribution, but we have no plan on this feature.
